mixer/cmd/config.go

```diff
@@ -73,6 +73,17 @@ var configSetCmd = &cobra.Command{
 	the existence of the provided property, but will not validate the value provided.`,
 	Args: cobra.ExactArgs(2),
 	Run: func(cmd *cobra.Command, args []string) {
+		// Look for the value in mixer.state first
+		var ms config.MixState
+		err := ms.Load("")
+		if err == nil {
```

Comment:
if err != nil it sounds like mixer.state doesn't exist, which is an error with mixer now right? I'm trying to re-think this logic a bit, along the lines of
```
	if err != nil {
		fail(err)
	}
	if err = config.SetProperty(&ms, args[0], args[1]) == nil {
		return
	}
```

Reply:
Then I don't get the separation between mixer.state and builder.conf.. which one matters? Might as well have a for loop that picks the first available config file in the directory at that point, which doesn't make sense. We chose to have 2 separate confs for a reason right, since it's a failure for either to not exist now.

Reply:
I am also confused about these configuration files and what they should contain, especially mixer.state. The mixer.state file name doesn't make me think of a configuration file, but rather a file that contains information about the current state of the mix. It would make sense to me if mixer.state was read by mixer to gather information about the current mix and updated by mixer commands that change the values in mixer.state.

Reply:
The idea behind the introduction of `mixer.state` was to have a single file that included values that are either set by mixer or that could change between mixer versions, while builder.conf would be a "base recipe" with values provided by the user. The plan is to have files like mixver, upstreamver and other auxiliary files alike to be part of mixer.state. The main reason why this change has not happened yet is that the file was introduced right before we started working closely with devops and our focus changed to keep the process stable. Since changing those files also require updating devops release process, the change was put on hold.

Also note that `mixer.state` was never intended to have a get/set command, I always saw it as something that would be solely manipulated internally by mixer. But again, devops...

Reply:
Also, there was a discussion not long ago about were the OFFLINE property should live. If it is in `mixer.state`, you can see it as "The mix is currently offline", while if it is in `builder.conf` you can see it as "This is an offline mix". The answer is whether users would shift between modes or set and forget. Either way, both solutions seems fine.
Now, even if we move it to `builder.conf` we still need a way to change values inside `mixer.state` because devops wants a way to set format without manually editing the file (and format is a value that for normal users should not change directly).

Reply:
@rchiossi I get what you're saying above but it doesn't really answer my question, we have a separation of content between confs, FORMAT is not parsed from builder.conf now, so missing mixer.state is an error, which mixer will try to solve by just picking a format for you. It has to fail if someone tries SetProperty() with format and mixer.state isn't there. Or am I misunderstanding?

Reply:
Considering a better future where `mixer.state` is not auto-generated if it is missing (which is what we want), if mixer.state is missing and the user tries to set FORMAT, it will fail since Mix.FORMAT will not be found.